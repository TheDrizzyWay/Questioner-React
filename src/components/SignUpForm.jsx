import React from 'react';

const SignUpForm = () => {
    return (
        <div id="wrapper">
            <form id="signup_form">
                <fieldset>
                    <legend>Signup Form</legend>
                    <div>
                        <input type="text" id="firstname" placeholder="First Name" maxLength="24" required />
                    </div>
                    <div>
                        <input type="text" id="lastname" placeholder="Last Name" maxLength="24" required />
                    </div>
                    <div>
                        <input type="email" id="email" placeholder="Email" required />
                    </div>
                    <div>
                        <input type="password" id="password" placeholder="Password" maxLength="18" required />
                    </div>
                    <div className="small">
                  Your password should be between six and 18 characters long.
                    </div>
                    <div>
                        <input type="password" id="password_confirmation" placeholder="Confirm Password"
                            maxLength="18" required />
                    </div>
                    <div>
                        <input type="text" id="username" placeholder="Username" maxLength="24" required />
                    </div>
                    <div>
                        <input type="tel" id="phonenumber" placeholder="Phone Number" maxLength="24" required />
                    </div>
                    <input type="submit" id="submit" value="Signup"/>
                </fieldset>
            </form>
        </div>
    );
};

export default SignUpForm;
