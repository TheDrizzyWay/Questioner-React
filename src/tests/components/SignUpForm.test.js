import React from 'react';
import { shallow } from 'enzyme';
import SignUpForm from '../../components/SignUpForm';

const props = {
    auth: {
        isLoading: false
    },
    history: {
        push: jest.fn()
    },
    signUp: jest.fn(),
    clearError: jest.fn()
};
let wrapper;

describe('<App />', () => {
    it('Should render component successfully', () => {
        wrapper = shallow(<SignUpForm.WrappedComponent {...props} />);
        expect(wrapper.length).toEqual(1);
    });

    test('Should maintain existing snapshot', () => {
        wrapper = shallow(<SignUpForm.WrappedComponent {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('should clear errors on focus', () => {
        wrapper = shallow(<SignUpForm.WrappedComponent {...props} />);
        wrapper.find('input').at(0).simulate('focus', { target: { id: 'firstname' } });
        expect(props.clearError).toHaveBeenCalled();
    });

    test('should correctly update state on input change', () => {
        wrapper = shallow(<SignUpForm.WrappedComponent {...props} />);
        wrapper.find('input').at(0).simulate('change', { target: { value: 'myname' } });
        wrapper.find('input').at(1).simulate('change', { target: { value: 'lastname' } });
        wrapper.find('input').at(2).simulate('change', { target: { value: 'user@email.com' } });
        wrapper.find('input').at(3).simulate('change', { target: { value: 'password' } });
        wrapper.find('input').at(4).simulate('change', { target: { value: 'password' } });
        wrapper.find('input').at(5).simulate('change', { target: { value: 'username' } });
        wrapper.find('input').at(6).simulate('change', { target: { value: '08011112222' } });
        wrapper.find('form').simulate('submit', { preventDefault: () => {} });
        expect(props.signUp).toHaveBeenCalled();
    });
});
