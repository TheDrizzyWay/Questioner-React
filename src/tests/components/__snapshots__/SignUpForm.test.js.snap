// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUpForm /> Should maintain existing snapshot 1`] = `ShallowWrapper {}`;
